R/descriptives.R

```diff
@@ -227,13 +202,11 @@ DescriptivesInternal <- function(jaspResults, dataset, options) {
       jaspResults[["stemAndLeaf"]]$position <- 11
     }
 
-    numericOrFactorVariables <- Filter(function(var) .descriptivesIsNumericColumn(dataset.factors, var), variables)
-
-    if (length(variables) > 0L) {
+    if (sum(variableTypes == "scale") > 0L) {
       .descriptivesStemAndLeafTables(
         container = jaspResults[["stemAndLeaf"]],
-        dataset   = if (makeSplit) splitDat.factors else dataset.factors,
-        variables = numericOrFactorVariables,
+        dataset   = if (makeSplit) splitDat else dataset,
+        variables = variables[variableTypes == "scale"],
```

Comment:
maybe just create

```r
scaleVariables    <- variables[variableTypes == "scale"]
nonScaleVariables <- variables[variableTypes != "scale"]
```

at the top, and then do everything based on the length of those variables? Right now there is a lot of repetition, so this might be a bit easier to read.